samples/nrf9160/gps/src/main.c

```diff
@@ -58,19 +57,23 @@ static const char *const at_commands[] = {
 	AT_ACTIVATE_GPS
 };
 
-static int                   gnss_fd;
-static char                  nmea_strings[10][NRF_GNSS_NMEA_MAX_LEN];
-static uint32_t                 nmea_string_cnt;
+static struct nrf_modem_gnss_pvt_data_frame  last_pvt;
+#ifdef CONFIG_SUPL_CLIENT_LIB
+static struct nrf_modem_gnss_agps_data_frame last_agps;
+static bool agps_requested;
+#endif
+static uint64_t fix_timestamp;
+static bool new_data;
+static bool gnss_blocked;
+static bool got_fix;
```

Comment:
Should these flags be `volatile`, as they're changed in ISR and used in other threads?

Reply:
Made these (and `agps_requested`) volatile.

Reply:
During refactoring I removed most of these. Only `gnss_blocked` remains, that's now `volatile`.